navigator_helpers/tasks/evaluation/evaluation.py

```diff
@@ -0,0 +1,585 @@
+# Script with all the classes and tests implemented
+from typing import Dict, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+from pandas import Series
+from pandas.core.dtypes.common import is_numeric_dtype
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics.pairwise import cosine_similarity
+
+from navigator_helpers.llms.llm_suite import GretelLLMSuite
+from navigator_helpers.tasks.base import BaseTaskSuite
+from navigator_helpers.tasks.prompt_templates import load_prompt_template_suite
+from navigator_helpers.tasks.text_to_code import utils
+
+llm_as_a_judge_prompts = load_prompt_template_suite("llm_as_a_judge")
+
+
+class BaseEvaluationTaskSuite(BaseTaskSuite):
+    """
+    Base class for evaluation task suites, which include a set of generic evaluation tasks :
+        1. Row uniqueness
+        2. Feature cardinality
+        3. Feature distribution
+        4. num_words per record
+        5. LLM-as-a-critic evaluation based on a generic dataset rubric (e.g. data quality, relevance, correctness, readability)
+    """
+
+    def __init__(
+        self,
+        llm_suite: GretelLLMSuite,
+        dataset: pd.DataFrame,
+        code_lang: Optional[str] = None,
+        eval_kwargs: Optional[Dict[str, str]] = None,
+    ) -> None:
+        super().__init__(llm_suite)
+        self.dataset = dataset
+        self.llm_suite = llm_suite
+        self.code_lang = code_lang
+        self.eval_kwargs = eval_kwargs
+        self.output_dataset = None
+
+    def _determine_column_data_type(self, column: Series) -> str:
+        """
+        Similar to https://github.com/Gretellabs/monogretel/blob/110622b0c8131f0c1a1443a8e25722a92857c3d2/python/src/gretel_core/harbor/artifacts/analyzers/field_features.py#L208C1-L281C20
+
+        Between the NavFT definition and the SQS Report definition of categorical variables,
+        here we adopt the SQS definition, where the key difference is the percentage of maximum allowed unique
+        values stays a constant 10% as dataset size increases, compared to the NavFT definition where that percentage
+        shrinks as the dataset size gets really large, like 1 million records.
+        The reasoning here is that the number of unique values, eg. data seeds,
+        should proportionally increase as the dataset size increases.
+        """
+        # If on average each entry has more than one space, we consider it a text field
+        _TEXT_FIELD_AVG_SPACE_COUNT_THRESHOLD = 1
```

Comment:
this value is set to 2 in the [similar method](https://github.com/Gretellabs/monogretel/blob/e23ada6b9de6fa6c1178d184eb183e0a32cbaaf6/python/src/gretel_core/harbor/artifacts/analyzers/field_features.py#L24) in the SQS report. Ideally we should try to have parity between the two, but I worry that if we leave too many columns into the "Other" type here that we don't know what to do about. 

In my mind, things in "Other" are mainly ID columns and datetimes that we don't really want to analyze on. A threshold of 1 should catch it. Please chime in if you have thoughts one way or another